### PR TITLE
chore: Fix CI and modules, and add tests (SC-3087)

### PR DESCRIPTION
**Pull request by deluca-mike**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,10 +10,15 @@ jobs:
           command: |
             GIT_SSH_COMMAND="ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no" git clone git@github.com:maple-labs/math.git .
             git checkout $CIRCLE_BRANCH
+      - run:
+          name: Build and test contracts
+          command: |
+            GIT_SSH_COMMAND="ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no" git submodule update --init --recursive
+            ./test.sh -c ./config/ci.json
```
Comment (lucas-manuel) [inline .circleci/config.yml]: `./test.sh`

File: Makefile [hunk 2]
```diff
@@ -1,4 +1,5 @@
-all    :; dapp build
-clean  :; dapp clean
-test   :; dapp test
-deploy :; dapp create Math
+prod    :; ./build.sh -c ./config/prod.json
+dev     :; ./build.sh -c ./config/dev.json
+ci      :; ./test.sh
```
Comment (lucas-manuel) [inline Makefile]: Can get rid of these

File: config/dev.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{
+  "language": "Solidity",
```
Comment (lucas-manuel) [inline config/dev.json:2]: Can get rid of this file

File: contracts/test/SafeMathUint.t.sol [hunk 4]
```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity 0.6.11;
+
+import { DSTest } from "../../modules/ds-test/src/test.sol";
+
+import { SafeMathUint } from "../SafeMathUint.sol";
+
+contract SafeMathIntTest is DSTest {
+
+    function test_toUint256Safe(uint256 x) public {
```
Comment (lucas-manuel) [inline contracts/test/SafeMathUint.t.sol]: `test_toInt256Safe`

File: contracts/test/SafeMathUint.t.sol [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        if (x > uint256(type(int256).max)) return;
+
+        assertEq(SafeMathUint.toInt256Safe(x), int256(x));
+    }
+
+    function testFail_toUint256Safe(uint256 x) public {
```
Comment (lucas-manuel) [inline contracts/test/SafeMathUint.t.sol]: `testFail_toInt256Safe`

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Small changes

File: config/dev.json (hunk 3 above)
Comment (deluca-mike) [reply config/dev.json:2]: Still need it to test if code compiles and whatnot. I like to keep it because it's not useless.

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**